framework/include/auxkernels/MortarNodalAuxKernel.h

```diff
@@ -10,14 +10,16 @@
 #pragma once
 
 #include "AuxKernel.h"
-#include "MortarInterface.h"
+#include "MortarConsumerInterface.h"
+#include "MortarExecutorInterface.h"
 
 /**
  * Base class for creating new nodally-based mortar auxiliary kernels
- *
  */
 template <typename ComputeValueType>
-class MortarNodalAuxKernelTempl : public AuxKernelTempl<ComputeValueType>, protected MortarInterface
+class MortarNodalAuxKernelTempl : public AuxKernelTempl<ComputeValueType>,
+                                  public MortarExecutorInterface,
+                                  protected MortarConsumerInterface
```

Comment:
The aux kernel gets to be consumer and executor. But the mortar constraint is only consumer because those duties are split with other classes?

Reply:
Correct. `ComputeMortarFunctor` is the object that executes the loop through the mortar mesh